proto/cmp/services/activity/v1alpha/activity_types.proto

```diff
@@ -0,0 +1,235 @@
+
+  // Supplier code from the Inventory System for this activity response.
+  string supplier_unit_code = 4;
+
+  // An identifier for external sessions, aiding in tracking and continuity across
+  // sessions.
+  string external_session_id = 5;
+
+  // Activity Category Code
+  string category_code = 6;
+
+  // Category name which describes the category_code
+  string category_name = 7;
+
+  // Type Code
+  string type_code = 8;
+
+  // Type name which describes type_code
+  string type_name = 9;
+
+  // Supplier specific code, matching the supplier code in ProductList and ProductInfo
+  // messages
+  cmp.types.v1alpha.SupplierProductCode supplier_code = 10;
+
+  // Supplier Name
+  string supplier_code_name = 11;
+
+  // product_code
+  cmp.types.v1alpha.ProductCode product_code = 12;
+
+  // Duration range (min, max) of the activity in minutes
+  cmp.types.v1alpha.DurationRange duration_range = 13;
+
+  // Time given for confirmation before the offer expires
+  cmp.types.v1alpha.Duration max_confirmation_duration = 14; 
+
+  // Allow Free Sale
+  bool allow_free_sale = 15;
+
+  // The ability to confirm instantly
+  bool instant_confirmation = 16;
+
+  // Can be delivered instantly
+  bool instant_delivery = 17;
+
+  // Check availability
+  bool availability_required = 18;
+  
+  // Availability Type
+  string availability_type = 19;
+
+  // Delivery Formats; such as QR Code, NFT, Ticket, ... etc
+  repeated string delivery_formats = 20;
+
+  // Delivery Methods; such as Email, SMS, Post, ... etc
+  repeated string delivery_methods = 21;
+
+  // Redemption method
+  string redemption_method = 22;
+
+}
+
+
+message ActivityUnitInfo {
+  // Schedule
+  cmp.types.v1alpha.DateTimeRange schedule = 1; 
+
+  // Description of the activity unit
+  string descriptive_text = 2;
+
+  // Unit Code
+  string unit_code = 3;
+
+  // Unit Code Description
+  string unit_code_description = 4;
+
+}
+
+
+message ActivityServiceInfo {
+  // Description of the Service Info
+  string descriptive_text = 1;
+  
+  // Service Code
+  string service_code = 2;
+
+  // Service Code Description
+  string service_code_description = 3;
+
+  // Included items in the activity service
+  repeated string included = 4;
+
+  // Encluded items in the activity service
+  repeated string excluded = 5;
+}
+
+message ActivityZoneInfo {
+  // Sub Zone Info
+  ActivitySubZoneInfo sub_zone_info = 1;
+
+  // Unit Code
+  string unit_code = 2;
+}
+
+message ActivitySubZoneInfo {
+  // Geo tree type, representted by Country, Region, and City_or_Resort.
+  repeated cmp.types.v1alpha.GeoTree ref_points = 1;
+
+  // operation time shows the pick-up and drop-off information
+  repeated OperationTime operation_time = 2;
+
+  // Unit Code
+  string unit_code = 3;
+}
+
+message OperationTime {
+  // pick-up and drop-off information about location and time
+  repeated cmp.services.activity.v1alpha.PickupDropoffEvent pickup_dropoff = 1;
+
+  // Unit Code
+  string unit_code = 2;
+}
```

Comment:
As far as I can see from the example json, each sub info has a single "operation time" which itself contains a list of "pickup dropoff events" inside. 

Is it correct to have repeated OperationTime and also repeated PickupDropoffEvent inside each?

Reply:
Operation time does not need to be repeated

Reply:
in the json each sub_info contains a list of "operation time" and in each operation_time there is a list of pickupdropoffevents.

Reply:
we removed the operation time wrapper